### PR TITLE
Query multiple ids

### PR DESCRIPTION
**Pull request by e0ipso**
For performance reasons we should allow clients to requests multiple entities at once.

Also our output structure should allow adding metadata to the response without polluting the response scope itself, hence wrapping it into a "data" object even if there is only one resource.

### CODE REVIEW

File: plugins/restful/RestfulEntityBase.php [hunk 1]
```diff
@@ -82,8 +82,8 @@
       // POST
       \RestfulInterface::POST => 'createEntity',
     ),
-    '\d+' => array(
-      \RestfulInterface::GET => 'viewEntity',
+    '^(\d+,)*\d+$' => array(
```
Comment (amitaibu) [inline plugins/restful/RestfulEntityBase.php:85]: maybe worth adding a comment with examples (e.g. `api/v1/articles/1 or api/v1/articles/1,2`)

File: plugins/restful/RestfulEntityBase.php [hunk 2]
```diff
@@ -765,6 +794,7 @@ public function getListAddHateoas(&$return, &$ids){
       // "next" page.
       array_pop($ids);
     }
+    $return['count'] = count($ids);
```
Comment (amitaibu) [inline plugins/restful/RestfulEntityBase.php]: Do we really need this count -- isn't it just a count of $result['data'], which can be achieved client side?

**Comment by amitaibu** [comment]
I really like this PR. Lets get the minor comments, and get it in. I hope people already using RESTful won't be too angry at us for changing the structure ;)

**Comment by e0ipso** [comment]
@amitaibu people already using RESTful are going to **love** what I'm working on right in #127. It'll give overriding capabilities on the output format.

File: plugins/restful/RestfulEntityBase.php (hunk 2 above)
Comment (e0ipso) [inline plugins/restful/RestfulEntityBase.php]: Nah, we don't. We should be putting the real count here (not the paginated count). But we can have a separate issue for that.

Comment (e0ipso) [inline plugins/restful/RestfulEntityBase.php]: I will remove this.

**Comment by e0ipso** [comment]
@amitaibu feedback implemented. Thanks for your suggestions.

**Comment by amitaibu** [comment]
Merged, thanks.

**Pull request closed without merging**